lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -1703,12 +1722,58 @@ class MailboxDashBoardController extends ReloadableController {
     EmailRequest emailRequest,
     CreateNewMailboxRequest? mailboxRequest
   ) {
-    consumeState(_sendEmailInteractor.execute(
-      session,
-      accountId,
-      emailRequest,
-      mailboxRequest: mailboxRequest
-    ));
+    if (PlatformInfo.isWeb) {
+      consumeState(_sendEmailInteractor.execute(
+        session,
+        accountId,
+        emailRequest,
+        mailboxRequest: mailboxRequest
+      ));
+    } else {
+      if (networkConnectionController.isNetworkConnectionAvailable()) {
+        consumeState(_sendEmailInteractor.execute(
+          session,
+          accountId,
+          emailRequest,
+          mailboxRequest: mailboxRequest
+        ));
+      } else {
+        final sendingEmail = emailRequest.toSendingEmail(_uuid.v1(), mailboxRequest: mailboxRequest);
+        _storeSendingEmail(accountId, session.username, sendingEmail);
+      }
```

Comment:
how about if all the case of network, we will call `_storeSendingEmai`?

Reply:
why call when there is network? If so, will we be sent the same email again when the network is changed!

Reply:
I think no, when network is connected, this action will be triggered after delay initial, so that, we remove from the sending queue is ok :))

Reply:
but if the task run will raise a notification, I think no need to call this in the case network connected